compiler/luci/pass/src/helpers/LayerInfoMap.cpp

```diff
@@ -22,6 +22,134 @@
+    case luci::CircleOpcode::CIRCLECUSTOMOUT:
+    {
+      const auto custom_out = loco::must_cast<CircleCustomOut *>(node);
+      return loco::must_cast<luci::CircleNode *>(custom_out->input());
+    }
+    case luci::CircleOpcode::CIRCLEIFOUT:
+    {
+      const auto if_out = loco::must_cast<CircleIfOut *>(node);
+      return loco::must_cast<luci::CircleNode *>(if_out->input());
+    }
+    case luci::CircleOpcode::CIRCLENONMAXSUPPRESSIONV4OUT:
+    {
+      const auto non_max_sus_v4_out = loco::must_cast<CircleNonMaxSuppressionV4Out *>(node);
+      return loco::must_cast<luci::CircleNode *>(non_max_sus_v4_out->input());
+    }
+    case luci::CircleOpcode::CIRCLENONMAXSUPPRESSIONV5OUT:
+    {
+      const auto non_max_sus_v5_out = loco::must_cast<CircleNonMaxSuppressionV5Out *>(node);
+      return loco::must_cast<luci::CircleNode *>(non_max_sus_v5_out->input());
+    }
+    case luci::CircleOpcode::CIRCLESPLITOUT:
+    {
+      const auto split_out = loco::must_cast<CircleSplitOut *>(node);
+      return loco::must_cast<luci::CircleNode *>(split_out->input());
+    }
+    case luci::CircleOpcode::CIRCLESPLITVOUT:
+    {
+      const auto splitv_out = loco::must_cast<CircleSplitVOut *>(node);
+      return loco::must_cast<luci::CircleNode *>(splitv_out->input());
+    }
+    case luci::CircleOpcode::CIRCLETOPKV2OUT:
+    {
+      const auto top_kv2_out = loco::must_cast<CircleTopKV2Out *>(node);
+      return loco::must_cast<luci::CircleNode *>(top_kv2_out->input());
+    }
+    case luci::CircleOpcode::CIRCLEUNIQUEOUT:
+    {
+      const auto unique_out = loco::must_cast<CircleUniqueOut *>(node);
+      return loco::must_cast<luci::CircleNode *>(unique_out->input());
+    }
+    case luci::CircleOpcode::CIRCLEUNPACKOUT:
+    {
+      const auto unpack_out = loco::must_cast<CircleUnpackOut *>(node);
+      return loco::must_cast<luci::CircleNode *>(unpack_out->input());
+    }
+    case luci::CircleOpcode::CIRCLEWHILEOUT:
+    {
+      const auto while_out = loco::must_cast<CircleWhileOut *>(node);
+      return loco::must_cast<luci::CircleNode *>(while_out->input());
+    }
+    case luci::CircleOpcode::CIRCLEBIDIRECTIONAL_SEQUENCE_LSTM_OUT:
+      throw std::runtime_error("Unsupported op now");
+    default:
+      return nullptr;
+  }
+}
+
+bool check_layer_info_equal(LayerInfo &left, LayerInfo &right)
+{
+  return left.dtype == right.dtype and left.granularity == right.granularity;
+}
+
+void add_multi_output_node(LayerInfoMap &info_by_name, LayerInfo &layer_info,
+                           luci::CircleNode *node)
+{
+  const auto succs_nodes = loco::succs(node);
+  auto name = node->name();
+
+  if (info_by_name.find(name) != info_by_name.end())
+  {
+    // Check that all outputs have equal dtype and granularity
+    for (const auto succs_node : succs_nodes)
+    {
+      const auto succs_circle_node = loco::must_cast<luci::CircleNode *>(succs_node);
+      name = succs_circle_node->name();
```

Comment:
Is it ok to update name here?

Reply:
Yes, we update name for all output nodes (for multiple output node) and check this name in the map.

Reply:
Ah, I missed the return statement at the end of the if statement.

Could you make this as a new variable? For example, `succ_name`? `name` is used in too many places in this function :)

Reply:
Done